[Fan Ruoyu] iP

frrrrry opened the pull request:
# Paimonbot

> “Let’s explore the area ahead of us later!” -Paimon

<img src="https://i.pinimg.com/1200x/dc/d3/6d/dcd36d823ad3a11355b2aca4cad6f365.jpg" width=200 height=200 />

Paimon frees your mind of having to remember things you need to do. It'S,
* text-based
* easy to learn
* ~~fast~~ SUPER FAST to use
* _not_ annoying ʘ‿ʘ

## All you need to do is, 

1. Download it from [here](https://github.com/frrrrry/ip.git)
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you

**AND ITS FREE!!!!**

## Features

- [x] Managing tasks
- [x] Managing deadlines
- [x] ReMINDERS
- [ ] daily commissions... 

If you are a Java programmer, you can use it to practice Java too. Here's the <code>main</code> method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

frrrrry committed:
Rename Chatbot

frrrrry committed:
Add echo features

frrrrry committed:
Add list

frrrrry committed:
Add checkbox feature

frrrrry committed:
Add Todos, Events and Deadlines

frrrrry committed:
Update test files

frrrrry committed:
Add error handling

frrrrry committed:
Add delete feature

frrrrry committed:
Update error handling

frrrrry committed:
Add enums for commands

frrrrry committed:
Add save feature

frrrrry committed:
Change format of date and time

frrrrry committed:
Add more classes to make code more OOP

frrrrry committed:
Change variable names

frrrrry committed:
Create new packages

frrrrry committed:
Merge branch 'add-gradle-support'

frrrrry committed:
Test gradle

frrrrry committed:
Update gradle build

frrrrry committed:
Add JUnit tests for classes in task package

frrrrry committed:
Add javadoc comments

frrrrry committed:
Edit code to conform to coding standards

frrrrry committed:
Add find keyword feature in chatbot

iyioon commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,64 @@
+package command;
+
+import duke.Ui;
```
I like how there is a consistent ordering of import statements. 👍

iyioon commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,41 @@
+public class Event extends Task {
+    protected LocalDate startDate;
+    protected LocalTime startTime;
+    protected String start;
```
Is there a better way to name start to better show its usage?

iyioon commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,64 @@
+                String[] dateTimeArr = by.split(" ");
+                LocalDate byDate = LocalDate.parse(dateTimeArr[0]);
+                LocalTime byTime = LocalTime.parse(dateTimeArr[1]);
+                Deadline d = new Deadline(description, byDate, byTime);
```
Is there a reason why you used `d` instead of other naming conventions?

iyioon commented on src/main/java/command/EventCommand.java:
```diff
@@ -0,0 +1,72 @@
+                String[] endArr = end.split(" ");
+                LocalDate endDate = LocalDate.parse(endArr[0]);
+                LocalTime endTime = LocalTime.parse(endArr[1]);
+                Event e = new Event(description, startDate, startTime, endDate, endTime);
```
Perhaps a more intuitive variable name here?

iyioon commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,56 @@
+    public static Command parse(String fullCommand) {
+        String command = fullCommand.split(" ")[0];
+        Keyword k = Keyword.valueOf(command.toUpperCase());
+        switch (k) {
```
I like how you followed the convention for switch statement. Looks like you have modified the format settings in your IDE 👍

iyioon reviewed (commented):
LGTM. All files are formatted correctly and it was nice to see not making common mistakes such as switch statement. Maybe you can improve variable names.

frrrrry committed:
Fix checkstyle errors

lihongguang00 commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,41 @@
+ * the chatbot.
+ */
+public abstract class Command {
+    private TaskList taskList;
```
Perhaps you could consider using the access modifier `protected` instead so your child classes can access it without storing another copy in them?

lihongguang00 commented on src/main/java/duke/Keyword.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * The Enums used to represent the main command keywords of the chatbot.
+ */
+public enum Keyword {
```
I like your use of enum

lihongguang00 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+    public static Command parse(String fullCommand) {
+        String command = fullCommand.split(" ")[0];
+        Keyword k = Keyword.valueOf(command.toUpperCase());
+        switch (k) {
```
I like your use of switch statement instead of multiple if-elseif.

lihongguang00 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,146 @@
+                char taskType = input.charAt(1);
+                boolean isDone = input.charAt(4) == 'X';
+                String description = input.split("] ")[1];
+                if (taskType == 'T') {
```
Perhaps you could consider using a switch statement for consistency?

frrrrry committed:
Add JavaFx configurations

lihongguang00 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,92 @@
+     * Prints out the welcome message of the chatbot.
+     */
+    public void showWelcome() {
+        String logo = " ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣀⣤⣤⣶⣶⣤⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀ \n"
```
i like this logo lol

lihongguang00 reviewed (commented):
I like the structure of your classes, although more can be done to abstract out common attributes, but overall it was a pleasant IP to review!

frrrrry committed:
Create mock GUI

frrrrry committed:
Create first draft of GUI for paimonbot

frrrrry committed:
Fix checkstyle errors

frrrrry committed:
Update GUI welcome message

Paimonbot GUI shows welcome message when user inputs 'hello' command.

This causes a blank GUI to be shown when user launches the chatbot which
can be very dull.

The initialize method in MainWindow now calls the dialogContainer to create
a new dialogBox with the paimonImage and the welcome message to be printed
out when the program is ran.

Showcasing a welcome message when chatbot is launched is more vibrant compared
to just having a blank space.

frrrrry committed:
Include assertion statements

frrrrry committed:
Merge pull request #1 from frrrrry/branch-A-Assertions

Add assertion statements

frrrrry committed:
Create new constructors for task classes

frrrrry committed:
Improve code quality

frrrrry committed:
Merge pull request #2 from frrrrry/branch-A-CodeQuality

Improve code quality

frrrrry committed:
Implement sort feature

Sort feature is implemented for C-Sort extension.

New commands are added into chatbot to sort deadlines and events using
the "sort deadlines" and "sort events" commands respectively.

Instead of printing the message string in the TaskList methods, it is now printed
from the methods in the Ui class.

It makes more sense for the Ui to return message strings instead of the
TaskList class since the TaskList class handles the operations on the ArrayLists
of Tasks.

The error handling for sort commands are not yet implemented.

frrrrry committed:
Merge pull request #3 from frrrrry/branch-BCD-Extension

Implement C-Sort Extension

frrrrry committed:
Update header comments

frrrrry committed:
Add close chatbot feature

frrrrry committed:
Update User Guide in README

frrrrry committed:
Add Ui.png

frrrrry committed:
Edit User Guide

frrrrry committed:
Merge pull request #5 from frrrrry/branch-A-UserGuide

Update the User Guide for PaimonBot
